### PR TITLE
lvgl:add lv_display_remove_event_cb_with_user_data

### PR DESCRIPTION
**Pull request by jianglianfang**
### Description of the feature or fix

Add lv_display_remove_event_cb_with_user_data becuase in many cases we don’t know the index corresponding to the event.

### Checkpoints
- [x] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<module_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: src/display/lv_display.h [hunk 1]
```diff
@@ -452,6 +452,8 @@ lv_event_dsc_t * lv_display_get_event_dsc(lv_display_t * disp, uint32_t index);
  */
 bool lv_display_delete_event(lv_display_t * disp, uint32_t index);
 
+uint32_t lv_display_remove_event_cb_with_user_data(lv_display_t * disp, lv_event_cb_t event_cb, void * user_data);
```
Comment (kisvegabor) [inline src/display/lv_display.h:462]: Please add comments

**Review by kisvegabor — COMMENTED** [review]
We have the index based approach to make LVGL better work with MicroPython, however for C user it's really useful.

Could you add the same for `lv_obj` and `lv_indev`?

File: src/display/lv_display.h (hunk 1 above)
Comment (jianglianfang) [reply src/display/lv_display.h:462]: OK, done

Comment (jianglianfang) [reply src/display/lv_display.h:462]: sure, I will add it for lv_obj and lv_indev soon in another commit

File: src/display/lv_display.c [hunk 2]
```diff
@@ -699,6 +699,25 @@ bool lv_display_delete_event(lv_display_t * disp, uint32_t index)
...
 
+uint32_t lv_display_remove_event_cb_with_user_data(lv_display_t * disp, lv_event_cb_t event_cb, void * user_data)
+{
+    LV_ASSERT_NULL(disp);
+
+    uint32_t event_cnt = lv_display_get_event_count(disp);
+    uint32_t i;
+    uint32_t removed_count = 0;
+
+    for(i = 0; i < event_cnt; i++) {
```
Comment (kisvegabor) [inline src/display/lv_display.c]: It just came to my mind that it won't work well if there are multiple events to delete after each other. Let's say we want to remove all `event_C_*` from this list:
0. event_A_1
1. event_B_1
2. event_C_1
3. event_C_2
4. event_C_3

When C_1 is deleted with i==2 the list will change like this:
0. event_A_1
1. event_B_1
2. event_C_2
3. event_C_3

In the next iteration `i`  will be `3`, so C_2 is skipped. 

A simple solution would be something like `if(i > 0) i--` after each deletion, or start from the end of the list.

Comment (jianglianfang) [reply src/display/lv_display.c]: Yeah, I think I might have overlooked that.

**Review by kisvegabor — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**